ImageMapClass.php

```diff
@@ -19,12 +19,16 @@ public function redcap_data_entry_form($project_id, $record, $instrument, $event
                 $keyLabelCodeMap[$fieldname]['script'] = "try{".$jsCall.'}catch(e) {console.warn(e);}' ;
```

Comment:
Just a question: yesterday, I also took a shot at trying to solve the issue, and I was trying to come up with a solution that would function in REDCap versions before v12.0.0, between v12.0.0 and v12.2.10 and after v12.2.10. I came up with the solution below (requiring no changes to the .js files, and only minimal changes to the config.json), but haven't thoroughly tested yet. Do you think this could also work? If yes, do you see advantages/disadvantages linked to both solutions?

As of line 19:
```php
                if ($redcap_version >= '12.2.10') {
                    $keyLabelCodeMap[$fieldname]['script'] = "module.afterRender(function() {try{" . $jsCall . '}catch(e) {console.warn(e);}})';
                } elseif ($redcap_version >= '12.0') {
                    $keyLabelCodeMap[$fieldname]['script'] = "REDCap.MultiLanguage.onLangChanged(function() {try{" . $jsCall . '}catch(e) {console.warn(e);}})';
                } else {
                    $keyLabelCodeMap[$fieldname]['script'] = "try{" . $jsCall . '}catch(e) {console.warn(e);}';
                }
            }
        }
        
        $this->initializeJavascriptModuleObject();
        echo '<script>';
        print file_get_contents(__DIR__ . "/imagemapfunctions.js");
        print file_get_contents(__DIR__ . "/imagemap.js");
        print "\n";
        print 'var module = ' . $this->getJavascriptModuleObjectName();
        print "\n";
        $this->activateMapScripts($keyLabelCodeMap);
        echo '</script>';
```